src/renderer/stores/sequencer.ts

```diff
@@ -76,23 +141,360 @@ export const useSequencerStore = create<State & Actions>()(
+        // Load the next sequence and run it
+        let nextIdx = idx + 1;
+        while (state.sequences[nextIdx].runable === false) {
+          nextIdx = nextIdx + 1;
+          if (nextIdx === state.sequences.length) {
+            return;
+          }
+        }
+        loadSequence(nextIdx, state);
+        sender(testSequenceRunRequest(state.testSequenceStepTree));
+        state.isLocked = true;
+      }),
+
+    addNewSequence: (
+      project: TestSequencerProject,
+      elements: TestSequenceElement[],
+    ) =>
+      set((state) => {
+        // Check if name is unique
+        const idx = state.sequences.findIndex(
+          (seq) => seq.project.name === project.name,
+        );
+        if (idx !== -1) {
+          return;
+        }
+        // Save this sequence
+        state.sequences.push({
+          project: project,
+          tree: createTestSequenceTree(elements),
+          elements: elements,
+          testSequenceUnsaved: false,
+          status: "pending",
+          runable: true,
+        });
+      }),
+
+    removeSequence: (name) =>
+      set((state) => {
+        // Check if sequence exists
+        console.log("removeSequence", name);
+        const idx = state.sequences.findIndex(
+          (seq) => seq.project.name === name,
+        );
+        if (idx === -1) {
+          return;
+        }
+        // Remove the sequence
+        state.sequences.splice(idx, 1);
+        // Check if the sequence is displayed
+        if (
+          state.testSequenceDisplayed !== null &&
+          state.testSequenceDisplayed.name === name
+        ) {
+          if (state.sequences.length !== 0) {
+            state.displaySequence(state.sequences[0].project.name);
+          } else {
+            clearSequencer(state);
+          }
+        }
+      }),
+
+    displaySequence: (name) =>
+      set((state) => {
+        // No Sequences, only steps
+        if (state.sequences.length === 0) {
+          return;
+        }
+        // Check if sequence already displayed
+        if (
+          state.testSequenceDisplayed !== null &&
+          state.testSequenceDisplayed.name === name
+        ) {
+          return;
+        }
+        // Check if the sequence exists
+        const idx = state.sequences.findIndex(
+          (seq) => seq.project.name === name,
+        );
+        if (idx === -1) {
+          return;
+        }
+        // save the current sequence if any
+        if (state.testSequenceDisplayed !== null) {
+          const oldIdx = state.sequences.findIndex(
+            (seq) => seq.project.name === state.testSequenceDisplayed?.name,
+          );
+          if (oldIdx !== -1) {
+            // Could be -1 if the sequence was removed
+            const oldSequence = containerizeCurrentSequence(oldIdx, state);
+            state.sequences[oldIdx] = oldSequence;
+          }
+        }
+        // Load the new sequence
+        loadSequence(idx, state);
+      }),
```

Comment:
same comment with the rest of all of these actions (and anywhere it happens in the rest of the store too), limit the scope of `set`.